src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -542,32 +542,26 @@ private UsernamePasswordAuthenticationToken loginAndSetUserData(String userName,
         grantedAuthorities.add(SecurityRealm.AUTHENTICATED_AUTHORITY);
 
         if (isNotBlank(groupsFieldName)) {
-            if(Strings.isNullOrEmpty(userInfoServerUrl)) {
-                if (containsField(idToken.getPayload(), groupsFieldName)) {
-                    LOGGER.fine("idToken contains group field name: " + groupsFieldName + " with value class:" + getField(idToken.getPayload(), groupsFieldName).getClass());
-                    @SuppressWarnings("unchecked")
-                    List<String> groupNames = (List<String>) getField(idToken.getPayload(), groupsFieldName);
-                    LOGGER.fine("Number of groups in groupNames: " + groupNames.size());
-                    for (String groupName : groupNames) {
-                        LOGGER.fine("Adding group from idToken: " + groupName);
-                        grantedAuthorities.add(new GrantedAuthorityImpl(groupName));
-                    }
-                } else {
-                    LOGGER.warning("idToken did not contain group field name: " + groupsFieldName);
+            if (!Strings.isNullOrEmpty(userInfoServerUrl) && containsField(userInfo, groupsFieldName)) {
```

Comment:
There's one little tiny thing is that might be confused. `Strings.isNullOrEmpty` can be replaced by `StringUtils.isBlank`. But it's beyond the current scope. You can skip it.